c_tests/tests/compile_multiblocks.c

```diff
@@ -0,0 +1,37 @@
+// Compiler:
+// Run-time:
+
+// Check that basic trace compilation works.
+// FIXME An optimising compiler can remove all of the code between start/stop
+// tracing.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  int cond = 2;
```

Comment:
If, for example, you make this test depend on `argc` rather than the constant `cond` you'll make it harder (not impossible!) for the compiler to optimise things away (though it will still potentially move them around).

Reply:
I'm not quite sure how we can pass things into those tests within our test suite. There currently doesn't seen to be an option for this.

Reply:
You don't have to pass anything in: `argc` is a "true" variable that the compiler can't guess is a constant.

Reply:
Fixed in 47f0ad0.